os/binfmt/libelf/libelf_cache.c

```diff
@@ -143,8 +138,20 @@ static off_t elf_cache_read_block(int filfd, uint16_t binary_header_size, FAR ui
 
 	binfo("filfd: %d block_number: %d binary_header_size: %d\n", filfd, block_number, binary_header_size);
 
-	/* Seek to location of 'block_number' block in elf file */
-	rpos = elf_cache_lseek_block(filfd, binary_header_size, block_number);
+	/* Seek to location of 'block_number' block in elf file for uncompressed elf */
+	if (elf_compress_type == COMPRESS_TYPE_NONE) {
+		rpos = elf_cache_lseek_block(filfd, binary_header_size, block_number);
+	}
+#ifdef CONFIG_COMPRESSED_BINARY
+	else {
+		if (elf_compress_type == CONFIG_COMPRESSION_TYPE) {
+			rpos = binary_header_size + (block_number * cache_blocks_size);
+		} else {
+			berr("No support for decompression of compression format %d of this binary\n", elf_compress_type);
```

Comment:
appropriate error could be returned here.

Reply:
done.